citesphere/src/main/java/edu/asu/diging/citesphere/web/validation/CitationConceptValidator.java

```diff
@@ -0,0 +1,51 @@
+package edu.asu.diging.citesphere.web.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import edu.asu.diging.citesphere.core.service.ICitationConceptManager;
+import edu.asu.diging.citesphere.model.IUser;
+import edu.asu.diging.citesphere.model.bib.ICitationConcept;
+import edu.asu.diging.citesphere.model.bib.IConceptType;
+import edu.asu.diging.citesphere.web.forms.CitationConceptForm;
+
+@Component
+public class CitationConceptValidator implements Validator {
+
+    @Autowired
+    private ICitationConceptManager conceptManager;
+
+    @Override
+    public boolean supports(Class<?> paramClass) {
+        return CitationConceptForm.class.equals(paramClass);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
+        CitationConceptForm conceptForm = (CitationConceptForm) target;
+        IUser user = (IUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (!errors.hasErrors()) {
+      	  if (conceptManager.getByUri(conceptForm.getUri()) != null) {
```

Comment:
Isn't this validator also used for editing concepts? Wouldn't this make any changes to an existing concept impossible?

Reply:
Also, two different users can create concepts with the same uri.